### PR TITLE
Provide indication of stale platforms and ability to update them

### LINKED ISSUE #7: As a WebWorks developer, I will receive a message in the GUI that will indicate if a project upgrade should be considered
Tied to internal Jira CLI issue:
https://jira.bbqnx.net/browse/BRWSR-15167
## Acceptance Criteria
- The user should have some indication when a project can be upgraded if the Cordova/WebWorks tools version is newer than the project's
- UX to be determined, initial proposal:
  - On opening of a project, check versions and provide an icon on the project bar to trigger an upgrade. This icon should be highlighted in some way 
  - Perhaps additionally, pop a toast on any project command invocation
- Regardless of UX, there should be way to trigger an upgrade.
  - When the user triggers the update, provide a confirmation dialog asking if they want to proceed, and provide a recommendation to backup their project prior to proceeding.
- Provide a progress bar to show progress and report success/failure of the operation.

### PR DESCRIPTION
**Pull request by jsoref**
The refresh icon is derived from this public domain icon:
 http://publicdomainvectors.org/en/free-clipart/Eco-water-recycling-vector-icon/6856.html
 -- it's actually only inspired by, this art was drawn by me using InkScape

Closes #7

### CODE REVIEW

File: lib/ui/views/updatesList.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+var UpdatesListView,
+    viewTemplate = {},
+    server = require("../server");
+
+function updatePlatform(config, platform, display) {
+    var newObj = true;
+}
```
Comment (rcruz) [inline lib/ui/views/updatesList.js]: Unused and mysterious function

**Comment by rcruz** [comment]
Functions well. Just need to match appearance with the rest of the page: font color #565e66, border of button #cccccc, button hover background color #cccccc

**Comment by rcruz** [comment]
@bryanhiggins @kwallis When upgrading a platform is possible, do we want to encourage the user to update? I'm thinking a highlight color on the upgrade notification will help with this. Thoughts?

**Comment by kwallis** [comment]
How risky is an upgrade? How often would it require code changes?

**Comment by jsoref** [comment]
@rcruz I think a highlight color is probably a good idea.

@kwallis within the **WebWorks** 1..2 cycle, we've occasionally messed up such that upgrading broke things (specifically Cordova's `exec` method is supposed to be **async**, but we had written a bunch of plugins that expected it to be **sync**, and when we fixed our `exec` to be **async** per spec, it broke all of our buggy plugins.) -- this was fixed before we shipped (well, for a weak definition of _fixed_).

OTOH, we also fixed a bunch of things for the 2.0 cycle such that in general an upgrade was a safe (to the best of our knowledge) action.

**Comment by bryanhiggins** [comment]
@kwallis The upgrade only affects the cordova-blackberry library, not the plugins, so it should never (or very rarely) require code changes. There is a story in the backlog to show plugins which have an update available.

**Comment by kwallis** [comment]
OK, let's highlight.

**Pull request closed without merging**